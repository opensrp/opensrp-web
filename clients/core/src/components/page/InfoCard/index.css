
#info-container{
    background-color: #FFFFFF;
    border: 1px solid #DADCE0;
    border-radius: 5px;
    box-shadow: 0 1px 3px 0 #D7D7E0;
    margin-top: 25px;
  
  }
  
  .info-header{
    margin: 15px;
    padding-bottom: 10px;
    border-bottom: 1px solid #DADCE0;
    color: #3A484F;
  }
  
  .info-body{
    margin: 0px 8px;
  }

  .table {
    border: none;
  }
  
  .react-table thead tr:first-child {
    display: none;
  }