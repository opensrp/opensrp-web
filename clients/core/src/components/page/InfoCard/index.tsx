import React from 'react';
import { Row, Col } from 'reactstrap';
import './index.css';

export interface InfoCardProps {
    title: string;
    children: React.ReactNode;
}
const InfoCard: React.FC<InfoCardProps> = (props: InfoCardProps) => {
    return (
        <div id="info-container">
            <Row className="info-header-bg">
                <Col className="info-header">
                    <span className="info-title">
                        <b>
                            <h5>{props.title}</h5>
                        </b>
                    </span>
                </Col>
            </Row>
            <Row>{props.children}</Row>
        </div>
    );
};

export default InfoCard;
