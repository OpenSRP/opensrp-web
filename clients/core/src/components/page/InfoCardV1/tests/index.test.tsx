import { shallow } from 'enzyme';
import React from 'react';
import InfoCard, { InfoCardProps, renderHeaderFunc, renderRowsFunc } from '..';

describe('components/page/InfoCard', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('renders without crashing', () => {
        const props: InfoCardProps = {
            renderHeader: renderHeaderFunc,
            renderRows: renderRowsFunc,
            visible: true,
            title: '',
            link: '',
            linkLable: '',
            rowData: [],
        };
        const component = shallow(<InfoCard {...props} />);
        expect(component).toMatchSnapshot();
    });
});
