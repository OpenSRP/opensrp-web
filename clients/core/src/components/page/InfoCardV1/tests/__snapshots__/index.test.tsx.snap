// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/page/InfoCard renders without crashing 1`] = `ShallowWrapper {}`;
